lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts

```diff
@@ -393,14 +400,62 @@ export default class ArcGisMapServerCatalogItem extends UrlMixin(
     return result;
   }
 
-  @computed
-  private get _currentImageryParts(): ImageryParts | undefined {
-    const dateAsUnix: string | undefined =
+  private getIntervalParamsFromDimensions() {
+    if (
+      this.modelDimensions.length < 1 ||
+      this.modelDimensions[0].selectedId === undefined
+    ) {
+      return undefined;
+    }
+    const selectedId = this.modelDimensions[0].selectedId;
+    if (this.modelDimensions.length < 1 && selectedId === undefined) {
+      return undefined;
+    } else {
```

Comment:
This function will be used to check if we want to do time window query. It is not to set traits.

Reply:
I probably didn't explain it well enough. The model dimensions are used for automatically setting trait values when the user selects a value from the dropdown. It is not meant to be parsed in code. There is also no need to parse it because you can directly access the `this.timeWindowDuration` trait value and expect it to be the value set by the user from the dropdown.

So I would remove this function and instead rewrite `getTimeParams` as something like:
```
private getTimeParams(): TimeParams {
    ...
    return {currentTime, timeWindowDuration: this.timeWindowDuration}
  }
```

Let me know if that makes sens, happy to chat.

Reply:
In the unit test, after updating model with this [json](https://github.com/TerriaJS/terriajs/blob/0d1f71b3f91fca42f1901d4c1423a98e11b5944b/test/Models/Catalog/esri/ArcGisMapServerCatalogItemSpec.ts#L514), the model has `modelDimensions` trait set only. How do I have `timeWindowDuration` and `timeUnit` set automatically without `modelDimensions` being parsed?

Reply:
See comment here - https://github.com/TerriaJS/terriajs/pull/6872#discussion_r1320925646